### PR TITLE
test(table, tablewithinputs, toast): add cypress tests for default

### PR DESCRIPTION
**Pull request by KatarzynaQA**
add cypress tests for default stories and refactor for classic stories related to table, toast and
table with inputs component

FE-2359

### Checklist

<del> - [ ] Release notes (Conventional Commits)  <del>
<del>  - [ ] Unit tests <del>  
- [x] Cypress automation tests

<del>  - [ ] Storybook added or updated <del>  
<del>  - [ ] Theme support <del>  
<del>  - [ ] Typescript `d.ts` file added or updated <del>  

### Additional context

add table, toast and table with input tests

### Testing instructions

run: ./node_modules/.bin/cypress run --spec './cypress/features/regression/table*.feature', './cypress/features/regression/toats*.feature' --reporter spec

### CODE REVIEW

File: cypress/features/regression/table.feature [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+  @positive
+  Scenario: I enable showPageSizeSelection
+    When I check paginate checkbox
+      And I check showPageSizeSelection checkbox
+    Then pageSize is visible
+
+  @positive
+  Scenario: I disable showPageSizeSelection
+    When I check paginate checkbox
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/table.feature]: fixed

File: cypress/features/regression/table.feature [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    Then pageSize is not visible
+
+  @positive
+  Scenario: I enable selectable
+    When I check selectable checkbox
+    Then rows are selectable
+
+  @positive
+  Scenario: I disable selectable
+    When I check selectable checkbox
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/table.feature]: fixed

File: cypress/features/regression/table.feature [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    Then row number <rowNumber> is highlighted
+    Examples:
+      | rowNumber |
+      | 1         |
+      | 2         |
+      | 3         |
+
+  @positive
+  Scenario Outline: Row <rowNumber> is not highlighted
+    When I check highlightable checkbox
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/table.feature]: fixed

File: cypress/features/regression/table.feature [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+  @positive
+  Scenario Outline: Set theme to <theme>
+    When I select theme to "<theme>"
+    Then theme on preview is "<theme>"
+    Examples:
+      | theme     |
+      | primary   |
+      | secondary |
+  # | tertiary       | will be only on default themes
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/table.feature]: fixed

File: cypress/features/regression/tableWithInputs.feature [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+  @positive
+  Scenario: I enable showPageSizeSelection
+    When I check paginate checkbox
+      And I check showPageSizeSelection checkbox
+    Then pageSize is visible
+
+  @positive
+  Scenario: I disable showPageSizeSelection
+    When I check paginate checkbox
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/tableWithInputs.feature]: fixed

File: cypress/features/regression/tableWithInputs.feature [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+    Then pageSize is not visible
+
+  @positive
+  Scenario: I enable selectable
+    When I check selectable checkbox
+    Then rows are selectable
+
+  @positive
+  Scenario: I disable selectable
+    When I check selectable checkbox
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/tableWithInputs.feature]: fixed

File: cypress/features/regression/tableWithInputs.feature [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+    Then row number <rowNumber> is highlighted
+    Examples:
+      | rowNumber |
+      | 1         |
+      | 2         |
+      | 3         |
+
+  @positive
+  Scenario Outline: Row <rowNumber> is not highlighted
+    When I check highlightable checkbox
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/tableWithInputs.feature]: fixed

File: cypress/features/regression/tableWithInputs.feature [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+
+  @positive
+  Scenario Outline: Set theme to <theme>
+    When I select theme to "<theme>"
+    Then theme on preview is "<theme>"
+    Examples:
+      | theme     |
+      | primary   |
+      | secondary |
+  # | tertiary       | will be only on default themes
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/tableWithInputs.feature]: fixed

File: cypress/features/regression/toast.feature [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+      | Sample text             |
+      | 1234567890              |
+      | áéíóú¿¡üñ               |
+      | !@#$%^*()_+-=~[];:.,?{} |
+      | ÄÖÜßäöüß                |
+      | <>                      |
+
+  @positive
+  Scenario: Enable open checkbox for a Toast component
+    When I uncheck open checkbox
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/toast.feature]: fixed

File: cypress/features/regression/toast.feature [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    Then Toast component is visible
+
+  @positive
+  Scenario: Disable open checkbox for a Toast component
+    When I uncheck open checkbox
+    Then Toast component is not visible
+
+  @positive
+  Scenario: Enable onDismiss checkbox for a Toast component
+    When I uncheck onDismiss checkbox
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/toast.feature]: fixed

File: cypress/support/step-definitions/table-steps.js [hunk 11]
```diff
@@ -26,6 +38,56 @@ When('row number {int} is not highlighted', (rowNumber) => {
...
+      .and('be.visible');
+  } else {
+    sortIcon(ONE).should('have.attr', 'data-element', 'sort_up')
+      .and('be.visible');
+  }
+});
+
+When('Country column is sorted in {string} order', (sortOrder) => {
+  if (sortOrder === 'desc') {
+    sortIcon(0).should('have.attr', 'data-element', 'sort_down')
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/support/step-definitions/table-steps.js]: fixed

File: cypress/support/step-definitions/table-steps.js [hunk 12]
```diff
@@ -26,6 +38,56 @@ When('row number {int} is not highlighted', (rowNumber) => {
...
+    rowNumbers(TWO).should('have.text', 'Albania');
+    rowNumbers(FOUR).should('have.text', 'Algeria');
+    rowNumbers(SIX).should('have.text', 'Andorra');
+    rowNumbers(EIGHT).should('have.text', 'Angola');
+  }
+});
+
+When('Code column is sorted in {string} order', (sortOrder) => {
+  if (sortOrder === 'desc') {
+    sortIcon(0).should('have.attr', 'data-element', 'sort_down')
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/support/step-definitions/table-steps.js]: fixed

File: cypress/support/step-definitions/table-steps.js [hunk 13]
```diff
@@ -34,11 +96,9 @@ Then('theme on preview is {string}', (theme) => {
   switch (theme) {
     case 'primary':
       tableHeader().should('have.css', 'background-color', 'rgb(51, 91, 109)');
-      tableHeader().should('have.css', 'color', 'rgb(255, 255, 255)');
       break;
     case 'secondary':
       tableHeader().should('have.css', 'background-color', 'rgb(204, 214, 218)');
-      tableHeader().should('have.css', 'color', 'rgb(0, 51, 73)');
       break;
     // will be only for a dafault theme
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/support/step-definitions/table-steps.js]: fixed

File: cypress/support/step-definitions/table-steps.js [hunk 14]
```diff
@@ -48,3 +108,52 @@ Then('theme on preview is {string}', (theme) => {
...
+      rowNumbers(ZERO).find('textarea');
+      break;
+    case 'date':
+      rowNumbers(ZERO).children().should('have.attr', 'data-component', 'date');
+      break;
+    default: throw new Error('There is only three input type of Table with inputs');
+  }
+});
+
+Then('Table classic totalRecords is set to {string} records', (totalRecords) => {
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/support/step-definitions/table-steps.js]: fixed

File: cypress/support/step-definitions/table-steps.js [hunk 15]
```diff
@@ -48,3 +108,52 @@ Then('theme on preview is {string}', (theme) => {
...
+  pagerSummary().invoke('text').should('contain', `${totalRecords}  records`);
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/support/step-definitions/table-steps.js]: fixed

File: cypress/features/regression/tableWithInputs.feature [hunk 16]
```diff
@@ -0,0 +1,200 @@
+Feature: Table With Inputs component
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/tableWithInputs.feature]: added in default story, I don't think so we need it here too

File: cypress/features/regression/table.feature [hunk 17]
```diff
@@ -0,0 +1,199 @@
+Feature: Table component
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/table.feature]: added event verification and test for action-toolbar

File: cypress/features/regression/table.feature [hunk 18]
```diff
@@ -0,0 +1,238 @@
...
+    Then <position> row has zebra striping
+    Examples:
+      | position |
+      | 0        |
+      | 1        |
+      | 4        |
+      | 5        |
+      | 8        |
+      | 9        |
+  @ignore
```
Comment (DlgSHi) [inline cypress/features/regression/table.feature]: space needed between 178 and 179 lines

File: cypress/features/regression/tableWithInputs.feature (hunk 8 above)
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/features/regression/tableWithInputs.feature]: still not uncommented

File: cypress/support/step-definitions/table-steps.js [hunk 19]
```diff
@@ -34,17 +97,126 @@ Then('theme on preview is {string}', (theme) => {
...
+    case 'textbox':
+      rowNumbers(ZERO).find('input').should('have.attr', 'data-element', 'input');
+      break;
+    case 'textarea':
+      rowNumbers(ZERO).find('textarea');
+      break;
+    case 'date':
+      rowNumbers(ZERO).children().should('have.attr', 'data-component', 'date');
+      break;
+    default: throw new Error('There is only three input type of Table with inputs');
```
Comment (DlgSHi) [inline cypress/support/step-definitions/table-steps.js]:
```suggestion
    default: throw new Error('There are only three input type of Table with inputs');
```

File: cypress/support/step-definitions/table-steps.js [hunk 20]
```diff
@@ -34,17 +97,126 @@ Then('theme on preview is {string}', (theme) => {
...
+    case 'last':
+      paginationButton(THREE).click();
+      break;
+    case 'previous':
+      paginationButton(ONE).click();
+      break;
+    case 'first':
+      paginationButton(ZERO).click();
+      break;
+    default: throw new Error('There is only four pagination buttons');
```
Comment (DlgSHi) [inline cypress/support/step-definitions/table-steps.js]:
```suggestion
    default: throw new Error('There are only four pagination buttons');
```

File: cypress/support/step-definitions/table-steps.js [hunk 21]
```diff
@@ -34,17 +97,126 @@ Then('theme on preview is {string}', (theme) => {
...
+Then('I click {string} pagination arrow', (arrow) => {
+  switch (arrow) {
+    case 'previousArrow':
+      nextArrow().parent().click();
+      previousArrow().parent().click();
+      break;
+    case 'nextArrow':
+      nextArrow().parent().click();
+      break;
+    default: throw new Error('There is only two pagination arrows');
```
Comment (DlgSHi) [inline cypress/support/step-definitions/table-steps.js]:
```suggestion
    default: throw new Error('There are only two pagination arrows');
```

File: cypress/support/step-definitions/table-steps.js [hunk 22]
```diff
@@ -34,17 +97,126 @@ Then('theme on preview is {string}', (theme) => {
...
+  }
+});
+
+When('I check checkbox on header', () => {
+  checkboxInHeader().eq(ZERO).click();
+});
+
+Then('Action Toolbar elemens are visible and has {string} color', (color) => {
```
Comment (DlgSHi) [inline cypress/support/step-definitions/table-steps.js]:
```suggestion
Then('Action Toolbar elemens are visible and have {string} color', (color) => {
```

Comment (KatarzynaQA) [reply cypress/support/step-definitions/table-steps.js]: fixed

File: cypress/support/step-definitions/table-steps.js (hunk 21 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/table-steps.js]: fixed

File: cypress/support/step-definitions/table-steps.js (hunk 20 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/table-steps.js]: fixed

File: cypress/support/step-definitions/table-steps.js (hunk 19 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/table-steps.js]: fixed

File: cypress/features/regression/table.feature (hunk 18 above)
Comment (KatarzynaQA) [reply cypress/features/regression/table.feature]: fixed

**Pull request merged**